sdk/keyvault/azure-security-keyvault-keys/migration_guide.md

```diff
@@ -0,0 +1,226 @@
+
+## Table of contents
+- [Migration benefits](#migration-benefits)
+- [Cross Service SDK improvements](#cross-service-sdk-improvements)
+- [Important changes](#important-changes)
+    - [Separate packages and clients](#separate-packages-and-clients)
+    - [Package names and namespaces](#package-names-and-namespaces)
+    - [Client constructors](#client-constructors)
+    - [Authentication](#authentication)
+    - [Common scenarios](#common-scenarios)
+        - [Async operations](#async-operations)
+        - [Create a key](#create-a-key)
+        - [Retrieve a key](#retrieve-a-key)
+        - [List properties of keys](#list-properties-of-keys)
+        - [Delete a key](#delete-a-key)
+        - [Perform cryptographic operations](#perform-cryptographic-operations)
+- [Additional samples](#additional-samples)
+
+## Migration benefits
+A natural question to ask when considering whether or not to adopt a new version or library is what the benefits of doing so would be. As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Java client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To try and improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [Java-specific guidelines](https://azure.github.io/azure-sdk/java_introduction.html) was also introduced to ensure that Java clients have a natural and idiomatic feel with respect to the Java ecosystem. Further details are available in the guidelines for those interested.
+
+### Cross Service SDK improvements
+The modern Key Vault Key client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as:
+
+- Using the new Azure Identity library to share a single authentication approach between clients.
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries.
+
+## Important changes
+### Separate packages and clients
+In the interest of simplifying the API, `azure-keyvault` and `KeyVaultClient` were split into separate packages and clients:
```

Comment:
Woah! This is the exact sentence @sadasant used as well! Did you folks talk about this???

:)

For JS, I recommended sticking to splitting of just the package. The split of client would then be obvious and the different clients are mentioned below any way

```suggestion
In the interest of simplifying the API for working with Key Vault keys, secrets and certificates, the `azure-keyvault`  package is split into separate packages.
```

Reply:
I based most of the contents besides the code samples and language-specific stuff con the Pyhton migration guides, since they have already been approved and look pretty good :)

Reply:
That explains everything!

Reply:
And keeps us as consistent as possible